packages/util/src/bytes.ts

```diff
@@ -225,6 +226,19 @@ export function bigIntToBuffer(num: bigint) {
   return toBuffer('0x' + num.toString(16))
 }
 
+/**
+ * Converts a {@link bigint} to a {@link Buffer} for devp2p rlp encodings
+ * excluding `0` as rlp has its own special encoding for the same
+ * see: https://github.com/ethereumjs/ethereumjs-monorepo/issues/2402
+ */
+export function devP2PBigIntToBuffer(input: bigint | number): BufferLike {
+  const inputNum = BigInt(input)
+  if (inputNum === 0n) {
+    return 0
+  } else {
+    return bigIntToBuffer(inputNum)
```

Comment:
I find it weird that `bigIntToBuffer` of `BigInt(0)` returns `<Buffer 00>`, it should IMO return the empty Buffer? Why did we choose to return this zero byte buffer?

Reply:
i could still find normal bigIntToBuffer handy in snap sync's jobs where origin we map from bigint ranges to buffers but yea usage is very limited.

Reply:
awesome will swap out the fn